seastar/src/config_and_start_seastar.rs

```diff
@@ -0,0 +1,293 @@
+        get_description(&self.opts)
+    }
+
+    /// Gets the `Options`' number of threads.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let opts = Options::new();
+    ///
+    /// assert_eq!(opts.get_smp(), num_cpus::get() as u32);
+    /// ```
+    pub fn get_smp(&self) -> u32 {
+        get_smp(&self.opts)
+    }
+
+    /// Sets the `Options`' name.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let mut opts = Options::new();
+    /// let name = "My Awesome Seastar App";
+    /// opts.set_name(name);
+    ///
+    /// assert_eq!(opts.get_name(), name);
+    /// ```
+    pub fn set_name(&mut self, name: &str) {
+        set_name(&self.opts, name);
+    }
+
+    /// Sets the `Options`' description.
+    ///
+    /// # Examples
+    ///
+    ///```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let mut opts = Options::new();
+    /// let description = "I love Seastar";
+    /// opts.set_description(description);
+    ///
+    /// assert_eq!(opts.get_description(), description);
+    /// ```
+    pub fn set_description(&mut self, description: &str) {
+        set_description(&self.opts, description);
+    }
+
+    /// Sets the `Options`' number of threads.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let mut opts = Options::new();
+    /// let smp = 42;
+    /// opts.set_smp(smp);
+    ///
+    /// assert_eq!(opts.get_smp(), smp);
+    /// ```
+    pub fn set_smp(&mut self, smp: u32) {
+        set_smp(&self.opts, smp);
+    }
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Options::new()
+    }
+}
+
+/// The object through which the contents of a `main` function would be ran in a Seastar app.
+/// Configurable through [`Options`].
+pub struct AppTemplate {
+    app: UniquePtr<app_template>,
+}
+
+impl AppTemplate {
+    /// Creates a Seastar app based on its configuration (`[seastar-rs::AppTemplate`]).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let app = AppTemplate::new_from_options(Options::default());
+    /// ```
+    pub fn new_from_options(opts: Options) -> Self {
+        AppTemplate {
+            app: new_app_template_from_options(&opts.opts),
+        }
+    }
+
+    /// Runs an app with a void callback (the output of which is always 0) and program arguments (argv).
+    /// The app is run on a separate thread.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let func = || { println!("{}", 42); };
+    ///
+    /// let app = AppTemplate::default();
+    /// let args = vec!["hello".to_owned()];
+    ///
+    /// assert_eq!(app.run_void(&args, func), 0);
+    /// ```
+    pub fn run_void(&self, args: &Vec<String>, func: fn()) -> i32 {
```

Comment:
Instead of passing a reference to a vector, it's slightly better to pass a slice. It allows for more use cases (you can easily convert a `&Vec` to `&[]`, but not the other way around.

Same suggestion applies to `run_int`.

Reply:
Agree.